src/components/networked-video-source.js

```diff
@@ -0,0 +1,87 @@
+/* global AFRAME, NAF, THREE */
+var naf = require('../NafIndex');
+
+AFRAME.registerComponent('networked-video-source', {
+
+  schema: {
+  },
+
+  dependencies: ['material'],
+
+  init: function () {
+    this.videoTexture = null;
+    this.video = null;
+    this.stream = null;
+
+    this._setMediaStream = this._setMediaStream.bind(this);
+
+    NAF.utils.getNetworkedEntity(this.el).then((networkedEl) => {
+      const ownerId = networkedEl.components.networked.data.owner;
+
+      if (ownerId) {
+        NAF.connection.adapter.getMediaStream(ownerId, "video")
+          .then(this._setMediaStream)
+          .catch((e) => naf.log.error(`Error getting media stream for ${ownerId}`, e));
+      } else {
+        // Correctly configured local entity, perhaps do something here for enabling debug audio loopback
+      }
+    });
+  },
+
+  _setMediaStream(newStream) {
+
+    if(!this.video) {
+      this.setupVideo();
+    }
+
+    if(newStream != this.stream) {
+      if (this.stream) {
+        this._clearMediaStream();
+      }
+
+      if (newStream) {
+        this.video.srcObject = newStream;
+
+        var playResult = this.video.play();
```

Comment:
Calling play is necessary if you reuse the video element and set again srcObject I guess. Ok. Otherwise when we create the video element it's with autoplay already.

Reply:
yes this also allow better auto-play support under certain cases if autoplay failed and user interacted then it will play.